screens/map/MapScreen.js

```diff
@@ -35,198 +36,158 @@ import {
 
 const snapPoints = [185, 325, 488];
 
-const deltas = {
-  latitudeDelta: 0.01,
-  longitudeDelta: 0.01,
-};
-
-export default class MapScreen extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      locationErrorMsg: null,
-      region: initialRegion,
-      stores: null,
-      store: null,
-      storeProducts: null,
-      showDefaultStore: false,
-    };
-  }
-
-  async componentDidMount() {
-    // We get current location first, since we need to use the lat/lon found in _populateIntitialStoresProducts
-    await this._findCurrentLocation();
-    await this._populateInitialStoresProducts();
-  }
+export default function MapScreen(props) {
+  const [locationPermissions, setLocationPermissions] = useState(null);
+  const [region, setRegion] = useState(initialRegion);
+  const [stores, setStores] = useState([]);
+  const [initialLocation, setInitialLocation] = useState(null);
+  const [currentStore, setCurrentStore] = useState(null);
+  const [storeProducts, setStoreProducts] = useState([]);
+  const [showDefaultStore, setShowDefaultStore] = useState(true);
+  const bottomSheetRef = React.useRef(null);
+  const mapRef = React.useRef(null);
+  const [isSorted, setSorted] = useState(false);
 
-  // TODO pretty high chance this should be either handled by navigation or `getDerivedStateFromProps`
-  // eslint-disable-next-line react/no-deprecated
-  componentWillReceiveProps(nextProps) {
-    const store = nextProps.route.params.currentStore;
-    const resetSheet = true;
-    store.distance = this.findStoreDistance(store);
-    this.changeCurrentStore(store, resetSheet);
-    const region = {
-      latitude: store.latitude,
-      longitude: store.longitude,
-      ...deltas,
-    };
-    this.setState({ region });
-  }
+  useCurrentLocation(setInitialLocation, setLocationPermissions);
+  useInitialStores(setStores);
+  useSortedStores(
+    locationPermissions,
+    stores,
+    setStores,
+    isSorted,
+    setSorted,
+    initialLocation
+  );
 
-  // Asks for permission if necessary, then gets current location
-  _findCurrentLocation = async () => {
-    const { status } = await Permissions.askAsync(Permissions.LOCATION);
-    // Error message not checked anywhere
-    if (status !== 'granted') {
-      this.setState({
-        locationErrorMsg: 'Permission to access location was denied',
-      });
-      Analytics.setUserProperty('location_permissions', 'denied');
-    } else {
+  // Only populate initial store products on first load
+  const initialLoadComplete = useRef(false);
```

Comment:
Some Medium article/forum told me to use a ref like this to make sure a `useEffect` only runs once. Would you say this is a common accepted way to do this?

Reply:
I'm not sure I have full context into "only runs once" — is there a limitation of an effect with an empty set of deps (`[]`)  that you've found? Otherwise that would be the option that I would use.

Reply:
I didn't go with the empty set of dependencies because I needed this to run until my initial conditions passed (locationPermissions set, stores loaded). I ended up using this workaround to make it keep checking if the conditions were met, then execute `populateInitialStoreProducts` once.